### PR TITLE
Учебный проект «Холст»

### PR DESCRIPTION
**Pull request by akzhar**
---
:mortar_board: [Учебный проект «Холст»](https://up.htmlacademy.ru/javascript/17/user/490939/tasks/3)

:boom: https://htmlacademy-javascript.github.io/490939-code-and-magick-17/2/

### CODE REVIEW

File: js/stats.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+}
+
+function drawCloudHeader(ctx, x, y, txtColor, font) {
+  ctx.fillStyle = txtColor;
+  ctx.font = font;
+  ctx.fillText('Ура вы победили!', x, y * 2);
+  ctx.fillText('Список результатов:', x, y * 3);
+}
+
+function drawBar(ctx, x, gistLowLevel, name, value, i, valueMax, txtColor) {
```
Comment (rkhaslarov) [inline js/stats.js]: Seems that here you're using too many args, it's the first sign of splitting the function

Comment (rkhaslarov) [reply js/stats.js]: or use objects =)

File: js/stats.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  return Math.random() * (max - min) + min;
+}
+
+window.renderStatistics = function (ctx, names, times) {
+
+  var valueMax = Math.round(Math.max.apply(null, times));
+  var gistLowLevel = CLOUD_HEIGHT - 20;
+  var contentX = CLOUD_X + 30;
+
+  drawCloud(ctx, CLOUD_X, CLOUD_Y, '#ffffff');
```
Comment (rkhaslarov) [inline js/stats.js]: '#ffffff' - constant?

File: js/stats.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+var CLOUD_Y = 10;
+var CLOUD_WIDTH = 420;
+var CLOUD_HEIGHT = 270;
+var BAR_WIDTH = 40;
+var BAR_GAP = 50;
+var GIST_MAX_HEIGHT = 150;
+var MY_BAR_COLOR = 'rgba(255, 0, 0, 1)';
+
+function drawCloud(ctx, x, y, bgColor) {
+  ctx.fillStyle = 'rgba(0, 0, 0, 0.7)';
```
Comment (rkhaslarov) [inline js/stats.js]: rgba(0, 0, 0, 0.7)' definitely should be a constant

File: js/stats.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+function drawCloudHeader(ctx, x, y, txtColor, font) {
+  ctx.fillStyle = txtColor;
+  ctx.font = font;
+  ctx.fillText('Ура вы победили!', x, y * 2);
+  ctx.fillText('Список результатов:', x, y * 3);
+}
+
+function drawBar(ctx, x, gistLowLevel, name, value, i, valueMax, txtColor) {
+
```
[the remark this answers is not in the document]
Comment (akzhar) [reply js/stats.js]: it makes sence))

File: js/stats.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+  var barX = x + (BAR_WIDTH + BAR_GAP) * i;
+  var barHeigth = value / valueMax * GIST_MAX_HEIGHT;
+  var barY = gistLowLevel - barHeigth;
+  var barColor = 'rgba(0, 0, 255, ' + getRandom(0.1, 1) + ')';
+
+  ctx.fillStyle = barColor;
+  if (name === 'Вы') {
```
[the remark this answers is not in the document]
Comment (akzhar) [reply js/stats.js]: done

File: js/stats.js [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+}
+
+function getRandom(min, max) {
+  return Math.random() * (max - min) + min;
+}
+
+window.renderStatistics = function (ctx, names, times) {
+
+  var valueMax = Math.round(Math.max.apply(null, times));
```
[the remark this answers is not in the document]
Comment (akzhar) [reply js/stats.js]: fixed

File: js/stats.js [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+  ctx.fillRect(x + 10, y + 10, CLOUD_WIDTH, CLOUD_HEIGHT);
+
+  ctx.fillStyle = bgColor;
+  ctx.fillRect(x, y, CLOUD_WIDTH, CLOUD_HEIGHT);
+}
+
+function drawCloudHeader(ctx, x, y, txtColor, font) {
+  ctx.fillStyle = txtColor;
+  ctx.font = font;
+  ctx.fillText('Ура вы победили!', x, y * 2);
```
[the remark this answers is not in the document]
Comment (akzhar) [reply js/stats.js]: done

File: js/stats.js [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+  ctx.fillText('Список результатов:', x, y * 3);
```
[the remark this answers is not in the document]
Comment (akzhar) [reply js/stats.js]: done

File: js/stats.js (hunk 1 above)
Comment (akzhar) [reply js/stats.js]: I used the object 'player' inside the 'for' cycle

File: js/stats.js (hunk 2 above)
Comment (akzhar) [reply js/stats.js]: I created constant object with colors

File: js/stats.js (hunk 3 above)
Comment (akzhar) [reply js/stats.js]: done

**Pull request merged**